lib/ts/recipe/emailpassword/utils.ts

```diff
@@ -253,9 +254,15 @@ export function getPasswordResetLink(input: {
     token: string;
     recipeId: string;
     tenantId: string;
+    userContext: any;
 }): string {
     return (
-        input.appInfo.websiteDomain.getAsStringDangerous() +
+        input.appInfo
+            .getOrigin({
+                request: getRequestFromUserContext(input.userContext),
```

Comment:
do not call getRequestFromUserContext here. Instead, make this function take a req | undefined. All the places that call getPasswordResetLink then need to pass in the request object if they have it, or then they can call getRequestFromUserContext if they can't get the request object (which should only be the case for index function exposed to the user).